server/src/main/java/io/druid/segment/realtime/firehose/WikipediaIrcDecoder.java

```diff
@@ -90,35 +90,59 @@ public WikipediaIrcDecoder(
     }
     this.namespaces = namespaces;
     this.geoIpDatabase = geoIpDatabase;
+    this.geoLookup = openGeoDB(geoIpDatabase);
+  }
 
-    File geoDb;
-    if (geoIpDatabase != null) {
-      geoDb = new File(geoIpDatabase);
-    } else {
+  private DatabaseReader openGeoDB(String configDbPath) {
+    RuntimeException lastEx = null;
+    for (int i=0; i<2; i++) {
+      File geoDb = configDbPath != null ? new File(configDbPath) : downloadGeoLiteDB();
       try {
-        String tmpDir = System.getProperty("java.io.tmpdir");
-        geoDb = new File(tmpDir, this.getClass().getCanonicalName() + ".GeoLite2-City.mmdb");
-        if (!geoDb.exists()) {
-          log.info("Downloading geo ip database to [%s]", geoDb);
-
-          FileUtils.copyInputStreamToFile(
-              new GZIPInputStream(
-                  new URL("http://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz").openStream()
-              ),
-              geoDb
-          );
-        }
+        return new DatabaseReader(geoDb);
       }
       catch (IOException e) {
-        throw new RuntimeException("Unable to download geo ip database [%s]", e);
+        lastEx = new RuntimeException("Unable to open geo ip lookup database at '"+geoDb.getAbsolutePath()+"', it may be corrupt.", e);
+        if (configDbPath != null) {
+          // if the config has an explicit db file path, then we assume we cannot delete it or re-download it.
+          throw lastEx;
+        }
+        log.warn(lastEx.getMessage()+" Attempting to re-download.", e);
+        geoDb.delete();
       }
     }
+
+    throw lastEx;
+  }
+
+  private File downloadGeoLiteDB() {
+    File geoDb;
     try {
-      geoLookup = new DatabaseReader(geoDb);
+      String tmpDir = System.getProperty("java.io.tmpdir");
+      geoDb = new File(tmpDir, this.getClass().getCanonicalName() + ".GeoLite2-City.mmdb");
+      if (!geoDb.exists()) {
```

Comment:
If it does exist, can you print an info message that you've found the downloaded file and the location it was found? That way the user is aware we have files residing on their system.